src/main/java/com/three_tech_solutions/slot_app/services/implementations/StudentServiceImpl.java

```diff
@@ -73,4 +76,14 @@ private boolean planTypeIsSpecificDay(CreateStudentRequest studentDTO) {
     private boolean planTypeIsBeginningOfMonth(CreateStudentRequest studentDTO) {
         return studentDTO.getPlanType().equals(PlanType.PRINCIPIO_DE_MES);
     }
+
+    public StudentDetailsResponse getStudentById(UUID studentId) {
+        Student student = studentRepository.findById(studentId)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Alumno no encontrado"));
+
+        List<Payment> payments = student.getPayments();
+
+        return studentMapper.toStudentDetailsResponse(student, payments);
```

Comment:
Siguiendo el comentario anterior, acá ya no haría falta pasarlo


```suggestion
        return studentMapper.toStudentDetailsResponse(student);
```